libs/blocks/merch/autoblock.js

```diff
@@ -20,3 +21,18 @@ export function postProcessAutoblock(autoblockEl) {
   localizePreviewLinks(autoblockEl);
   autoblockEl.querySelectorAll('.modal.link-block').forEach((blockEl) => loadBlock(blockEl));
 }
+
+export function decorateCardCtasWithA11y(card) {
```

Comment:
can you also rather move this function within postProcessAutoblock function? if it _needs_ to be executed after requestupdate & enable analytics, then maybe we can rediscuss. But on a semantic side, this is post process. On a code size, it will change much less files.

Reply:
right, but that work of waiting for current element or current element's cards to be ready and then decorate could be done by decorateCardCtasWithA11y itself

Reply:
Ok like this ?

```
function decorateCtasWithA11y(collection) {
  collection.querySelectorAll('merch-card').forEach((merchCard) => decorateCardCtasWithA11y(merchCard));
}
```

and in autoblock.js

```
export async function decorateCardCtasWithA11y(card) {
  await card.checkReady();
  card.querySelectorAll() ...
```